### ISSUE TITLE
Test explorer is not discovering mocha tests in the subdirectories

### ISSUE THREAD

**Issue by xugao** [Microsoft/nodejstools#1596]
##### Expected Behavior 
Test explorer should show all tests in sub-directories
##### Actual Behavior 
Only shows tests in the root directory

- NTVS Version: --
- Visual Studio Version: 2017
- Node.js Version: 6.9.1

##### Steps to Reproduce
1. I have a nodejs project with a file which its Testframework is set to Mocha and created a test suite/test in it. 
1. Import another JS/TS file that is in a different project
2. msbuild will compile and output the test file into a sub-folder instead of into the root
3. Test explorer could not discover it

**Comment by paulvanbrenk** [comment]
I'm not sure I understand your repro, can you create a sample project and put it on GitHub so I can take a look?

**Comment by paulvanbrenk** [comment]
I was unable to reproduce this.

**Comment by cb504** [comment]
I have the same issue: 
Microsoft Visual Studio Professional 2017, Version 15.4.4
Node.js Tools   1.4.10918.1

I saw in the "test" output window that VS was searching the file in the sub-directory "tests", but not finding any tests. I moved my test file up from "<root>\tests" directory to "<root>\"  and now I see my tests in the test explorer. 
If I add the test file to **both** directories, "<root>\test.js" AND "<root>\tests\test.js". Now, VS will discover both sets of tests, _but the files must contain the same tests_. This can be used as a workaround, I then  group tests by "namespace" and just run the tests in the sub-directory.

**Comment by Aishwarya-Sivaraman** [comment]
Hi, 
Even after following the above steps in VS 2017, Mocha Test is not getting discovered in my project.
1. I have installed mocha in the project within the solution. (My solution has 3 projects)
2. I have set the Test Framework in the properties window to Mocha. In fact adding a new file of type "Typescript Mocha Unit Test file" automatically sets.
3. I have bothe describe and it . Sample code below.
describe("Test Suite 1", () => {
    it("Test A", () => {
        assert.ok(true, "This shouldn't fail");
    });

    it("Test B", () => {
        assert.ok(1 === 1, "This shouldn't fail");
        assert.ok(false, "This should fail");
    });
});

Any idea why this behaviour?
